### ISSUE TITLE
Kernel 6.3.7-200.fc38.x86_64 support

### ISSUE THREAD

**Issue by EmergReanimator** [ivanovborislav/rtl8812au#3]
Hello Borislav,

Would you please be so kind to take a look at the build failure here?
Do you have any idea how to fix it?

```CC [M]  /var/lib/dkms/rtl8812au/5.13.6/build/hal/rtl8812a/rtl8812a_cmd.o
/var/lib/dkms/rtl8812au/5.13.6/build/os_dep/linux/ioctl_cfg80211.c: In function ‘rtw_cfg80211_ch_switch_notify’:
/var/lib/dkms/rtl8812au/5.13.6/build/os_dep/linux/ioctl_cfg80211.c:458:17: error: too few arguments to function ‘cfg80211_ch_switch_started_notify’
  458 |                 cfg80211_ch_switch_started_notify(adapter->pnetdev, &chdef, 0, 0, false);
      |                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /var/lib/dkms/rtl8812au/5.13.6/build/include/osdep_service_linux.h:98,
                 from /var/lib/dkms/rtl8812au/5.13.6/build/include/osdep_service.h:54,
                 from /var/lib/dkms/rtl8812au/5.13.6/build/include/drv_types.h:27,
                 from /var/lib/dkms/rtl8812au/5.13.6/build/os_dep/linux/ioctl_cfg80211.c:17:
./include/net/cfg80211.h:8392:6: note: declared here
 8392 | void cfg80211_ch_switch_started_notify(struct net_device *dev,
      |      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/var/lib/dkms/rtl8812au/5.13.6/build/os_dep/linux/ioctl_cfg80211.c:481:9: error: too few arguments to function ‘cfg80211_ch_switch_notify’
  481 |         cfg80211_ch_switch_notify(adapter->pnetdev, &chdef, 0);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~
./include/net/cfg80211.h:8375:6: note: declared here
 8375 | void cfg80211_ch_switch_notify(struct net_device *dev,
      |      ^~~~~~~~~~~~~~~~~~~~~~~~~
  CC [M]  /var/lib/dkms/rtl8812au/5.13.6/build/hal/rtl8812a/usb/usb_halinit.o
  CC [M]  /var/lib/dkms/rtl8812au/5.13.6/build/hal/rtl8812a/usb/rtl8812au_led.o
  CC [M]  /var/lib/dkms/rtl8812au/5.13.6/build/hal/rtl8812a/usb/rtl8812au_xmit.o
  CC [M]  /var/lib/dkms/rtl8812au/5.13.6/build/hal/rtl8812a/usb/rtl8812au_recv.o
  CC [M]  /var/lib/dkms/rtl8812au/5.13.6/build/hal/rtl8812a/usb/usb_ops_linux.o
  CC [M]  /var/lib/dkms/rtl8812au/5.13.6/build/hal/efuse/rtl8812a/HalEfuseMask8812A_USB.o
  CC [M]  /var/lib/dkms/rtl8812au/5.13.6/build/hal/rtl8812a/hal8812a_fw.o
  CC [M]  /var/lib/dkms/rtl8812au/5.13.6/build/hal/phydm/phydm_debug.o
make[2]: *** [scripts/Makefile.build:252: /var/lib/dkms/rtl8812au/5.13.6/build/os_dep/linux/ioctl_cfg80211.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [Makefile:2037: /var/lib/dkms/rtl8812au/5.13.6/build] Error 2```

**Comment by EmergReanimator** [comment]
In fact [aircrack-ng driver](https://github.com/aircrack-ng/rtl8812au) is available for 6.3.7;

**Comment by ivanovborislav** [comment]
Hi!
Successful compiled...
```
make ARCH=x86_64 CROSS_COMPILE= -C /lib/modules/6.3.7-060307-generic/build M=/home/vm/Desktop/git-test/rtl8812au  modules
make[1]: Entering directory '/usr/src/linux-headers-6.3.7-060307-generic'
warning: the compiler differs from the one used to build the kernel
  The kernel was built by: x86_64-linux-gnu-gcc-12 (Ubuntu 12.3.0-1ubuntu1) 12.3.0
  You are using:           gcc-12 (Ubuntu 12.1.0-2ubuntu1~22.04) 12.1.0
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/core/rtw_cmd.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/core/rtw_security.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/core/rtw_debug.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/core/rtw_io.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/core/rtw_ioctl_query.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/core/rtw_ioctl_set.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/core/rtw_ieee80211.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/core/rtw_mlme.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/core/rtw_mlme_ext.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/core/rtw_mi.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/core/rtw_wlan_util.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/core/rtw_vht.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/core/rtw_pwrctrl.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/core/rtw_rf.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/core/rtw_chplan.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/core/monitor/rtw_radiotap.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/core/rtw_recv.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/core/rtw_sta_mgt.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/core/rtw_ap.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/core/wds/rtw_wds.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/core/mesh/rtw_mesh.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/core/mesh/rtw_mesh_pathtbl.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/core/mesh/rtw_mesh_hwmp.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/core/rtw_xmit.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/core/rtw_p2p.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/core/rtw_rson.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/core/rtw_tdls.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/core/rtw_br_ext.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/core/rtw_iol.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/core/rtw_sreset.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/core/rtw_btcoex_wifionly.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/core/rtw_btcoex.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/core/rtw_beamforming.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/core/rtw_odm.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/core/rtw_rm.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/core/rtw_rm_fsm.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/core/rtw_ft.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/core/rtw_wnm.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/core/rtw_mbo.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/core/rtw_rm_util.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/core/efuse/rtw_efuse.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/core/rtw_roch.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/core/crypto/aes-internal.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/core/crypto/aes-internal-enc.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/core/crypto/aes-gcm.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/core/crypto/aes-ccm.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/core/crypto/aes-omac1.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/core/crypto/ccmp.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/core/crypto/gcmp.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/core/crypto/aes-siv.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/core/crypto/aes-ctr.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/core/crypto/sha256-internal.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/core/crypto/sha256.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/core/crypto/sha256-prf.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/core/crypto/rtw_crypto_wrap.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/core/rtw_swcrypto.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/os_dep/osdep_service.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/os_dep/linux/os_intfs.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/os_dep/linux/usb_intf.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/os_dep/linux/usb_ops_linux.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/os_dep/linux/ioctl_linux.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/os_dep/linux/xmit_linux.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/os_dep/linux/mlme_linux.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/os_dep/linux/recv_linux.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/os_dep/linux/ioctl_cfg80211.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/os_dep/linux/rtw_cfgvendor.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/os_dep/linux/wifi_regd.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/os_dep/linux/rtw_android.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/os_dep/linux/rtw_proc.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/os_dep/linux/nlrtw.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/os_dep/linux/rtw_rhashtable.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/os_dep/linux/ioctl_mp.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/hal/hal_intf.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/hal/hal_com.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/hal/hal_com_phycfg.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/hal/hal_phy.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/hal/hal_dm.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/hal/hal_dm_acs.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/hal/hal_btcoex_wifionly.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/hal/hal_btcoex.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/hal/hal_mp.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/hal/hal_mcc.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/hal/hal_hci/hal_usb.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/hal/led/hal_led.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/hal/led/hal_usb_led.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/hal/HalPwrSeqCmd.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/hal/rtl8812a/Hal8812PwrSeq.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/hal/rtl8812a/Hal8821APwrSeq.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/hal/rtl8812a/rtl8812a_xmit.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/hal/rtl8812a/rtl8812a_sreset.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/hal/rtl8812a/rtl8812a_hal_init.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/hal/rtl8812a/rtl8812a_phycfg.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/hal/rtl8812a/rtl8812a_rf6052.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/hal/rtl8812a/rtl8812a_dm.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/hal/rtl8812a/rtl8812a_rxdesc.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/hal/rtl8812a/rtl8812a_cmd.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/hal/rtl8812a/usb/usb_halinit.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/hal/rtl8812a/usb/rtl8812au_led.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/hal/rtl8812a/usb/rtl8812au_xmit.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/hal/rtl8812a/usb/rtl8812au_recv.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/hal/rtl8812a/usb/usb_ops_linux.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/hal/efuse/rtl8812a/HalEfuseMask8812A_USB.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/hal/rtl8812a/hal8812a_fw.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/hal/phydm/phydm_debug.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/hal/phydm/phydm_antdiv.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/hal/phydm/phydm_soml.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/hal/phydm/phydm_smt_ant.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/hal/phydm/phydm_antdect.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/hal/phydm/phydm_interface.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/hal/phydm/phydm_phystatus.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/hal/phydm/phydm_hwconfig.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/hal/phydm/phydm.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/hal/phydm/phydm_dig.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/hal/phydm/phydm_pathdiv.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/hal/phydm/phydm_rainfo.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/hal/phydm/phydm_dynamictxpower.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/hal/phydm/phydm_adaptivity.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/hal/phydm/phydm_cfotracking.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/hal/phydm/phydm_noisemonitor.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/hal/phydm/phydm_beamforming.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/hal/phydm/phydm_direct_bf.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/hal/phydm/phydm_dfs.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/hal/phydm/txbf/halcomtxbf.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/hal/phydm/txbf/haltxbfinterface.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/hal/phydm/txbf/phydm_hal_txbf_api.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/hal/phydm/phydm_adc_sampling.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/hal/phydm/phydm_ccx.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/hal/phydm/phydm_psd.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/hal/phydm/phydm_primary_cca.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/hal/phydm/phydm_cck_pd.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/hal/phydm/phydm_rssi_monitor.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/hal/phydm/phydm_auto_dbg.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/hal/phydm/phydm_math_lib.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/hal/phydm/phydm_api.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/hal/phydm/phydm_pow_train.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/hal/phydm/phydm_lna_sat.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/hal/phydm/phydm_pmac_tx_setting.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/hal/phydm/phydm_mp.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/hal/phydm/phydm_cck_rx_pathdiv.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/hal/phydm/halrf/halrf.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/hal/phydm/halrf/halrf_debug.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/hal/phydm/halrf/halphyrf_ce.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/hal/phydm/halrf/halrf_powertracking_ce.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/hal/phydm/halrf/halrf_powertracking.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/hal/phydm/halrf/halrf_kfree.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/hal/phydm/halrf/halrf_psd.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/hal/phydm/rtl8812a/halhwimg8812a_mac.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/hal/phydm/rtl8812a/halhwimg8812a_bb.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/hal/phydm/rtl8812a/halhwimg8812a_rf.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/hal/phydm/halrf/rtl8812a/halrf_8812a_ce.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/hal/phydm/rtl8812a/phydm_regconfig8812a.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/hal/phydm/rtl8812a/phydm_rtl8812a.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/hal/phydm/txbf/haltxbfjaguar.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/platform/platform_ops.o
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/core/rtw_mp.o
  LD [M]  /home/vm/Desktop/git-test/rtl8812au/8812au.o
  MODPOST /home/vm/Desktop/git-test/rtl8812au/Module.symvers
  CC [M]  /home/vm/Desktop/git-test/rtl8812au/8812au.mod.o
  LD [M]  /home/vm/Desktop/git-test/rtl8812au/8812au.ko
  BTF [M] /home/vm/Desktop/git-test/rtl8812au/8812au.ko
Skipping BTF generation for /home/vm/Desktop/git-test/rtl8812au/8812au.ko due to unavailability of vmlinux
make[1]: Leaving directory '/usr/src/linux-headers-6.3.7-060307-generic'
```
